### PR TITLE
Decorator configuration improvements

### PR DESCRIPTION
**Pull request by venkatajagannath**
Currently, ray configuration to the ray.task decorator can only be a static input.

This PR fixes that behavior and also allows users to provide the configuration at runtime.

We are also making the following updates --

- Documentation updates to mention that config can be static or dynamic
- Added a new example dag that generates a config and passing in the callable

### CODE REVIEW

File: example_dags/ray_taskflow_example_dynamic_config.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    catchup=False,
+    tags=["ray", "example"],
+)
+def ray_taskflow_dag():
+
+    @task
+    def generate_data():
+        return [1, 2, 3]
+
+    @ray.task(config=generate_config)
```
Comment (dukarc) [inline example_dags/ray_taskflow_example_dynamic_config.py:38]: Not sure this is dynamic .. for instance would we be able to reference the airflow context to build out a config that uses the dag_run_id?
```suggestion
    @ray.task(config=generate_config(context))
```

Comment (venkatajagannath) [reply example_dags/ray_taskflow_example_dynamic_config.py:38]: If its only for uniqueness of the cluster name, can we use any UUID?

File: example_dags/ray_taskflow_example_dynamic_config.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    return {
+        "conn_id": CONN_ID,
+        "runtime_env": {"working_dir": str(FOLDER_PATH), "pip": ["numpy"]},
+        "num_cpus": 1,
+        "num_gpus": 0,
+        "memory": 0,
+        "poll_interval": 5,
+        "ray_cluster_yaml": str(RAY_SPEC),
+        "xcom_task_key": "dashboard",
+    }
```
Comment (dukarc) [inline example_dags/ray_taskflow_example_dynamic_config.py:22]: this was just an earlier example .. i may use this approach to make it easy for team members to select from a few preformed specs

i.e. 

config = generate_config(type="gpu_task")
config = generate_config(type="memory_task")
...

this is a very simple example we might go further and spec out cpu/memory requirements, image etc to use for ray workers

Comment (venkatajagannath) [reply example_dags/ray_taskflow_example_dynamic_config.py]: made an update. please recheck

**Pull request closed without merging**